### PR TITLE
fix: fixed table rendering (html and gfm) and reorganised mdx plugins

### LINKED ISSUE #6002: Table is not rendered
### URL:

https://nodejs.org/en/docs/guides/debugging-getting-started#command-line-options

### Browser Name:

Mozilla Firefox

### Browser Version:

118.0.1

### Operating System:

Fedora Linux 37 (Workstation Edition)

### How to reproduce the issue:

1. What I did.

I opened the documentation page at "https://nodejs.org/en/docs/guides/debugging-getting-started#command-line-options".

2. What I expected to get.

The options table properly rendered.

3. What I actually got.

The options table was missing.

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

This PR fixes HTML tables and GFM (GitHub Flavoured Markdown) tables not rendering. It adds support to:

- GitHub Flavoured Markdown Tables (raw Markdown) (version 1.0.7 of the plugin, as 2.0.0 crashes `next-mdx-remote`) through a MDast Plugin (`mdast-util-gfm-table`)
- HTML Tables for plain Markdown (as MDX already parses "HTML" (actually JSX) tables) through a Rehype Plugin (`rehype-raw`)
  - This plugin should **only** be used with plain Markdown as MDX already uses it, and otherwise it creates conflicts.

This PR segments all the MDX Plugins into a dedicated file `next.mdx.js` to keep the `next.dynamic.mjs` file clean.

## Validation

All MDX/Markdown features should still be working as:

- Headings having IDs
- Headings having Anchor Tags
- Raw Links becoming Anchor Tags
- HTML tables being rendered
- Codeboxes still work correctly

## Related Issues

Closes #6002

### CODE REVIEW

File: next.mdx.mjs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        theme: shikiNordTheme,
+        defaultLang: 'plaintext',
+        getHighlighter: options =>
+          getHighlighter({ ...options, langs: SUPPORTED_LANGUAGES }),
+      },
+    ],
+  ];
+
+  if (fileExtension === 'md') {
+    // We add the rehype-raw at the too (first plugin) as it should parse raw HTML
```
Comment (rodion-arr) [inline next.mdx.mjs]: Probably some word is missed after `at the`

**Review by rodion-arr — APPROVED** [review]
Checked the reported link on preview site, looks fine
https://nodejs-org-git-fix-tables-and-reorganised-mdx-plugins-openjs.vercel.app/en/docs/guides/debugging-getting-started#command-line-options

File: next.mdx.mjs (hunk 1 above)
Comment (ovflowd) [reply next.mdx.mjs]:
```suggestion
    // We add this plugin at the top of the array as it is supposed to parse raw HTML
    // before any other plugins (such as adding headings, etc)
```

**Comment by ovflowd** [comment]
Hey @nodejs/nodejs-website I'm fast-tracking this due to being an existing bug and this being a "hot fix"

**Pull request merged**